### PR TITLE
pxe_boot: Make test exit once found TFTP packet

### PR DESCRIPTION
**Pull request by yanan-fu**
id: 1755711
Signed-off-by: Yanan Fu <yfu@redhat.com>

### CODE REVIEW

**Comment by yanan-fu** [comment]
@luckyh @PaulYuuu updated as we discussed before to make it more robust, please help check again, thanks!

File: generic/tests/pxe_boot.py [hunk 1]
```diff
@@ -22,12 +35,9 @@ def run(test, params, env):
...
     vm.verify_alive()
     timeout = int(params.get("pxe_timeout", 60))
-
     error_context.context("Snoop packet in the tap device", logging.info)
-    output = aexpect.run_fg("tcpdump -nli %s" % vm.get_ifname(),
-                            logging.debug, "(pxe capture) ", timeout)[1]
-
-    error_context.context("Analyzing the tcpdump result", logging.info)
-    if "tftp" not in output:
+    tcpdump_cmd = "tcpdump -nli %s" % vm.get_ifname()
```
[the remark this answers is not in the document]
Comment (yanan-fu) [reply generic/tests/pxe_boot.py]: @PaulYuuu  thanks, as we discussed,  keep more tcpdump message for debugging in case test failed.

**Comment by PaulYuuu** [comment]
I still recommend filtering out redundant messages such as arp, icmp, etc... They are not good for debugging.
Or `port bootps and tftp` can capture dhcp and tftp related messages.
Acked-by: Yihuang Yu <yihyu@redhat.com>

File: generic/tests/pxe_boot.py [hunk 2]
```diff
@@ -24,10 +24,14 @@ def run(test, params, env):
...
+    try:
+        tcpdump_process = aexpect.run_bg(command=tcpdump_cmd,
+                                         output_func=logging.debug,
+                                         output_prefix="(pxe capture) ")
+        if not tcpdump_process.read_until_output_matches(['tftp'],
+                                                         timeout=timeout):
+            test.fail("Couldn't find any TFTP packets after %s seconds" % timeout)
+        logging.info("Found TFTP packet")
+    finally:
+        tcpdump_process.kill()
```
Comment (luckyh) [inline generic/tests/pxe_boot.py]:
```python
try:
    tcpdump_process.kill()
except:
    pass
```

Comment (yanan-fu) [reply generic/tests/pxe_boot.py]: @luckyh make sense, updated, please help review again, thanks!

**Comment by luckyh** [comment]
LGTM, please provide the test result for the latest code.

**Comment by yanan-fu** [comment]
(1/1) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.1.x86_64.io-github-autotest-qemu.pxe_boot: PASS (24.66 s)

And debug.log:
```
01:22:25 DEBUG| (pxe capture) 01:22:25.222677 IP 10.73.227.183.37558 > 10.73.2.110.**tftp:**  40 RRQ "pxelinux.0" octet blksize 1432 tsize 0
01:22:25 DEBUG| (pxe capture) 01:22:25.225538 IP 10.73.2.110.47730 > 10.73.227.183.37558: UDP, length 27
01:22:25 INFO | Found TFTP packet
...
01:22:25 DEBUG| (pxe capture) (Process terminated with status 0)
```

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**